nautobot/extras/datasources/registry.py

```diff
@@ -46,7 +46,7 @@ def refresh_datasource_content(model_name, record, request, job_result, delete=F
                     job_result.log(
                         f"Error while refreshing {entry.name}: {exc}", level_choice=LogLevelChoices.LOG_FAILURE
                     )
-                    job_result.set_status(JobResultStatusChoices.STATUS_ERRORED)
+                    job_result.set_status(JobResultStatusChoices.STATUS_FAILURE)
```

Comment:
Does this file also need a TODO about removing the `set_status` calls?

Reply:
Actually I meant to remove all of those. Anywhere explicit status changes are being made must go away. We want Celery to make these decisions, along with any explicit calls to `JobResult.save()`, moving those to the `NautobotTask` class or the backend or a mix of both, depending.

Reply:
@jathanism Do you intend to do this in this PR or open a follow-on story?

Reply:
Follow-on as a part of the "make `run_job` a thing of the past" work.